src/main/java/Model/Product.java

```diff
@@ -2,13 +2,37 @@
 public class Product {
     private String name;
     private String description;
-    private Float value;
-    private Boolean isOnSale;
+    private Float regularValue;
+    private Boolean isDiscounted;
+    private Float discountedValue;
 
-public Product(String name, String description, Float value) {
+public Product(String name, String description, Float regularValue, Float discountedValue) {
     this.name = name;
     this.description = description;
-    this.value = value;
-    this.isOnSale = false;
+    this.regularValue = regularValue;
+    this.discountedValue = discountedValue;
+    this.isDiscounted = false;
+    }
+
+    public void discount(Integer percent) {
+    this.isDiscounted = true;
+    this.discountedValue = this.regularValue - (percent * this.regularValue);
```

Comment:
Percent is provided in the function as Integer, so percent will equal 20 or 50 or 10 etc.
It means that you can't calculate discountedValue as this.regularValue - (percent * this.regularValue).
You can do that for example like this: this.regularValue - (**percent/100** * this.regularValue).

Reply:
Can I just change percent into Double or Float?

Reply:
Then other programmer would need to check what that function actually does to be sure if he needs to pass 31.5 or 0.315.
Functions should be intuitive.

Reply:
done